controllers/cache_handler_test.go

```diff
@@ -138,11 +138,13 @@ var _ = Describe("Cache Handler verifications", func() {
 			Expect(err).To(BeNil())
 
 			By("ensuring created event was fired")
-			event := findEvent(disruption.CreationTimestamp.Time, v1beta1.Events[v1beta1.EventDisrupted], eventList.Items, targetPodA.Name)
-			Expect(event).ToNot(BeNil())
+			Eventually(func(g Gomega) {
+				event := findEvent(disruption.CreationTimestamp.Time, v1beta1.Events[v1beta1.EventDisrupted], eventList.Items, targetPodA.Name)
+				Expect(event).ShouldNot(BeNil())
+			}).Should(Succeed())
```

Comment:
Looking at the documentation, it seems we only have to create a func with the return type expected:

```suggestion
			Eventually(func() *v1.Event {
				return findEvent(disruption.CreationTimestamp.Time, v1beta1.Events[v1beta1.EventDisrupted], eventList.Items, targetPodA.Name)
			}).ShouldNot(BeNil())
```

If you still prefer the syntax with Gomega as parameter, [it seems you need](https://onsi.github.io/gomega/#category-3-making-assertions-eminem-the-function-passed-into-codeeventuallycode) to use `g` to perform `Expect`?

```suggestion
			Eventually(func(g Gomega) {
				event := findEvent(disruption.CreationTimestamp.Time, v1beta1.Events[v1beta1.EventDisrupted], eventList.Items, targetPodA.Name)
				g.Expect(event).ShouldNot(BeNil())
			}).Should(Succeed())
```
> Note that the assertions in the body of the polled function must be performed using the passed-in g Gomega object. If you use the global DSL expectations, Eventually will not intercept any failures and the test will fail.

Reply:
Ahh, I missed the `g.Expect`, thank you